### PR TITLE
define core protocol/port combinations

### PR DESCRIPTION
**Pull request by dprotaso**
Part of https://github.com/kubernetes-sigs/gateway-api/issues/1842

Unblocks - https://github.com/kubernetes-sigs/gateway-api/pull/1859

### Release Note
```release-note
Define Core supported Listener Port and Protocol Combinations.
```

### CODE REVIEW

**Comment by robscott** [comment]
Thanks @dprotaso! This probably needs codegen + a release note but otherwise LGTM. Will add a hold for consensus on this.

/hold
/cc @arkodg @howardjohn @sunjayBhatia @mlavacca @shaneutt @youngnick

**Review by sunjayBhatia — APPROVED** [review]

File: apis/v1beta1/gateway_types.go
Comment (arkodg) [inline apis/v1beta1/gateway_types.go:73]:
```suggestion
	// Port, and Protocol. Below combinations are considered Core and must be supported:
```

Comment (robscott) [reply apis/v1beta1/gateway_types.go:73]: Good catch, this is better. One tiny additional suggestion on top of this - we usually capitalize "MUST" in the API Spec.

**Review by mlavacca — APPROVED** [review]

**Comment by robscott** [comment]
Thanks @dprotaso!

/hold cancel
/lgtm
/approve

**Review by arkodg — APPROVED** [review]

**Pull request merged**
